lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -432,4 +460,3 @@ def on_update(self, proposed_object, ib_spec):
             update = ib_spec[key].get('update', True)
             if not update:
                 keys.add(key)
-        return dict([(k, v) for k, v in iteritems(proposed_object) if k not in keys])
```

Comment:
Kindly undo this change as this returns value from `on_update` function.

Reply:
@justjais done. I've added it back. It was unintentional.